ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteConnectorExecutor.java

```diff
@@ -107,11 +107,15 @@ private Tuple<Integer, Integer> calculateChunkSize(TextDocsInputDataSet textDocs
                 throw new IllegalArgumentException("no predict action found");
             }
             String preProcessFunction = predictAction.get().getPreProcessFunction();
-            if (preProcessFunction != null && !MLPreProcessFunction.contains(preProcessFunction)) {
-                // user defined preprocess script, this case, the chunk size is always equals to text docs length.
+            if (preProcessFunction == null) {
+                // default preprocess case, consider this a batch.
+                return Tuple.tuple(1, textDocsLength);
+            } else if (MLPreProcessFunction.TEXT_DOCS_TO_BEDROCK_EMBEDDING_INPUT.equals(preProcessFunction)
```

Comment:
Add some IT to cover this ?

Reply:
Do we have key&secret for bedrock IT?

Reply:
I think we can merge this first, current code missing a lot of remote inference IT cases and configuring key & secret doesn't seem a convenient approach when writing ITs, I'll create an issue to run mock servers based on the request size so that key & secret will not be an necessity of ITs.

Reply:
Check this class RestMLInferenceIngestProcessorIT.java